src/components/Popover.tsx

```diff
@@ -110,15 +133,13 @@ class Popover extends React.Component<any, any> {
             y: wrapperRect.top,
         }
 
-        /** @type {("left" | "right" | "top" | "bottom")[]} */
-        const positionsToTry =
-            position === 'auto'
-                ? ['top', 'right', 'bottom', 'left', 'top']
-                : position === 'vertical'
-                ? ['top', 'bottom']
-                : position === 'horizontal'
-                ? ['left', 'right']
-                : [position]
+        const positionsToTry = ((position === 'auto'
```

Comment:
`const positionsToTry: (RelativePosition | undefined)[] =`

`undefined ` here could be a mistake, is `position` really optional?

Reply:
Agreed, position can never be undefined.